test/integ_tests/test_device_creation.py

```diff
@@ -75,42 +75,55 @@ def test_get_devices_all():
 
 
 def test_device_enum():
-    provider_name_to_enum_map = {
-        "Amazon Braket": "Amazon",
-        "D-Wave Systems": "_DWave",
-        "IonQ": "IonQ",
-        "Oxford": "OQC",
-        "QuEra": "QuEra",
-        "Rigetti": "Rigetti",
-        "Xanadu": "_Xanadu",
-    }
-    device_name_to_enum_map = {
-        "SV1": "SV1",
-        "TN1": "TN1",
-        "dm1": "DM1",
-        "Advantage_system1.1": "_Advantage1",
-        "Advantage_system3.2": "_Advantage3",
-        "Advantage_system4.1": "_Advantage4",
-        "Advantage_system6.1": "_Advantage6",
-        "DW_2000Q_6": "_DW2000Q6",
-        "Harmony": "Harmony",
-        "Aria 1": "Aria1",
-        "Lucy": "Lucy",
-        "Aquila": "Aquila",
-        "Aspen-8": "_Aspen8",
-        "Aspen-9": "_Aspen9",
-        "Aspen-10": "_Aspen10",
-        "Aspen-11": "_Aspen11",
-        "Aspen-M-1": "_AspenM1",
-        "Aspen-M-2": "_AspenM2",
-        "Aspen-M-3": "AspenM3",
-        "Borealis": "_Borealis",
-    }
-    for device in AwsDevice.get_devices():
-        assert (
-            getattr(
-                getattr(Devices, provider_name_to_enum_map[device.provider_name]),
-                device_name_to_enum_map[device.name],
-            )
-            == device.arn
+    def get_provider_name(device: AwsDevice):
+        arn_provider = device.arn.split("/")[2]
+
+        # capitalize as in provider name
+        provider_primary_name = device.provider_name.split()[0]
+        if arn_provider == provider_primary_name.lower():
+            capitalized = provider_primary_name
+        else:
+            capitalized = arn_provider.upper()
+
+        # remove dashes
+        return capitalized.replace("-", "")
+
+    def get_device_name(device: AwsDevice):
+        arn_device_name = device.arn.split("/")[-1]
+
+        device_name = (
+            arn_device_name
+            .replace("_", "")
+            .replace("-", "")
+            .replace("system", "")
         )
+
+        if device.provider_name == "Amazon Braket":
+            device_name = device_name.upper()
+        return device_name
+
+    aws_devices = AwsDevice.get_devices()
+
+    # determine retired providers
```

Comment:
Comments like this usually hint that I should just move it to a function. I do that less for test functions, but here I think it would be useful.